### PR TITLE
Add feature flags recording demo for developer guide

### PR DESCRIPTION
**Pull request by JeromeJu**
/kind documentation

# Changes
This commit adds feature flags recording demo for developer guide.

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [n/a] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [n/a] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [n/a] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/developers/feature-versioning.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 The stability levels of features (feature versioning) are independent of CRD [API versioning](./api-versioning.md).
 
+You may follow the existing Tekton feature flags demo for detailed reference:
+- [Tekton Per-feature Flag Demo Slides](https://docs.google.com/presentation/d/1MAwBTKYUN40SZcd6om6LMw217TtNSppbmA8MatMjyjk/edit?usp=sharing&resourcekey=0-JY7-QhCrWJrzFgsFbJGROg)
```
Comment (Yongxuanzhang) [inline docs/developers/feature-versioning.md:6]: just want to double check people should have the permission to read this ?

Comment (JeromeJu) [reply docs/developers/feature-versioning.md:6]: Thanks for checking, yep i shared with tekton-dev@googlegroups.com and tekton-user@googlegroups.com

**Comment by JeromeJu** [comment]
The CI has been pretty weird 🤔

File: docs/developers/feature-versioning.md (hunk 1 above)
Comment (Yongxuanzhang) [reply docs/developers/feature-versioning.md:6]: 👍

**Review by Yongxuanzhang — APPROVED** [review]

**Comment by JeromeJu** [comment]
The notification to https://prow.tekton.dev/pr-history/?org=tektoncd&repo=pipeline&pr=7662 is correct
while for GHA it fails 🤔

**Comment by afrittoli** [comment]
> The notification to https://prow.tekton.dev/pr-history/?org=tektoncd&repo=pipeline&pr=7662 is correct while for GHA it fails 🤔

Something is wrong with GitHub status update tasks, the CI pipeline runs ok but the status is not reflected in GitHub

**Pull request merged**